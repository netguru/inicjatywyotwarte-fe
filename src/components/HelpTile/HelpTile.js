import React, { useState } from 'react'
import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUp'
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown'
import Collapse from '@material-ui/core/Collapse'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Container = styled.div`
  background: ${props => props.theme.lightBackgroundColor};
  border: 1px solid ${props => props.theme.resourceTileLayoutColor};
  border-radius: ${props => props.theme.borderRadiusSmall};
`

const Tile = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  cursor: pointer;
  padding: 1.3rem 2.5rem;
`

const HeaderText = styled.div`
  font-weight: 400;
  font-size: 1rem;
`

const HeaderArrow = styled.div`
  display: flex;
  align-items: center;
  transition: ${props => props.theme.buttonTransition};

  &:hover {
    box-shadow: ${props => props.theme.shadowButtonHover};
    border-radius: 50px;
  }
`

const Description = styled.div`
  padding: 1rem 2.5rem 1.3rem;
  font-weight: 300;
  font-size: 0.9rem;
  line-height: 1.5rem;
  color: ${props => props.theme.modalColorLight};
  text-align: justify;
`

const HelpTile = ({ headerText, descriptionText }) => {
  const [isOpen, setIsOpen] = useState(false)

  const toggleIsOpen = () => {
    setIsOpen(!isOpen)
  }

  return (
    <Container>
      <Tile
        id={headerText}
        aria-controls={`${headerText}sectionId`}
        aria-expanded={isOpen.toString()}
        aria-disabled={isOpen.toString()}
        onClick={toggleIsOpen}
      >
        <HeaderText>{headerText}</HeaderText>
        <HeaderArrow>
          {isOpen ? <ArrowDropUpIcon /> : <ArrowDropDownIcon />}
        </HeaderArrow>
      </Tile>
      <Collapse timeout={300} in={isOpen}>
        <Description
          id={`${headerText}sectionId`}
          role='region'
          aria-labelledby={headerText}
        >
          {descriptionText}
        </Description>
      </Collapse>
    </Container>
  )
}

export default HelpTile

HelpTile.propTypes = {
  headerText: PropTypes.string.isRequired,
  descriptionText: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
}
