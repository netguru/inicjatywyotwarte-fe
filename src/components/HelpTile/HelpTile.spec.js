import React from 'react'
import HelpTile from './HelpTile'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('HelpTile', () => {
  test('renders help til with proper ', () => {
    const wrapper = shallow(
      <HelpTile
        headerText="header text"
        descriptionText="description text"
      />
      );
    expect(wrapper.find('HeaderText').text()).toEqual('header text')
    expect(wrapper.find('Description').text()).toEqual('description text')
  });

  test('has dropdown arrow working', () => {
    const wrapper = shallow(
      <HelpTile
        headerText="header text"
        descriptionText="description text"
      />
    );
    expect(wrapper.find('ArrowDropDownIcon').length).toEqual(1);
    wrapper.find('Tile').simulate('click');
    expect(wrapper.find('ArrowDropUpIcon').length).toEqual(1);
  });

  test('match snapshot', () => {
    const wrapper = shallow(
      <HelpTile
        headerText="header text"
        descriptionText="description text"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


