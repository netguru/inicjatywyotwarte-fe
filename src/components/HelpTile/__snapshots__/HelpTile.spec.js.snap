// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpTile match snapshot 1`] = `
<styled.div>
  <Tile
    aria-controls="header textsectionId"
    aria-disabled="false"
    aria-expanded="false"
    id="header text"
    onClick={[Function]}
  >
    <HeaderText>
      header text
    </HeaderText>
    <HeaderArrow>
      <ArrowDropDownIcon />
    </HeaderArrow>
  </Tile>
  <WithStyles(ForwardRef(Collapse))
    in={false}
    timeout={300}
  >
    <Description
      aria-labelledby="header text"
      id="header textsectionId"
      role="region"
    >
      description text
    </Description>
  </WithStyles(ForwardRef(Collapse))>
</styled.div>
`;
